core/src/java/show/ezkz/casprzak/core/tools/functiongenerators/HermitePolynomial.java

```diff
@@ -0,0 +1,106 @@
+package show.ezkz.casprzak.core.tools.functiongenerators;
+
+import show.ezkz.casprzak.core.config.Settings;
+import show.ezkz.casprzak.core.functions.GeneralFunction;
+import show.ezkz.casprzak.core.functions.binary.Pow;
+import show.ezkz.casprzak.core.functions.commutative.Product;
+import show.ezkz.casprzak.core.functions.commutative.Sum;
+import show.ezkz.casprzak.core.functions.endpoint.Constant;
+import show.ezkz.casprzak.core.functions.endpoint.Variable;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.lang.Math.floor;
+import static java.lang.Math.pow;
+import static show.ezkz.casprzak.core.tools.MiscTools.factorial;
+import static show.ezkz.casprzak.core.tools.defaults.DefaultFunctions.*;
+
+/**
+ * The methods in {@link HermitePolynomial} deal with <a href="https://en.wikipedia.org/wiki/Hermite_polynomials#Hermite_functions">Hermite Polynomials</a>.
+ */
+public class HermitePolynomial {
+
+	private static final Map<Integer, GeneralFunction> cache = new HashMap<>();
+
+	/**
+	 * Returns the nth Hermite polynomial with the default variable defined with {@link Settings#singleVariableDefault}
+	 * @param n the nth polynomial
+	 * @return nth Hermite polynomial
+	 */
+	public static GeneralFunction hermitePolynomial(int n) {
+		if (Settings.cacheDerivatives && cache.containsKey(n))
+			return cache.get(n);
+
+		GeneralFunction[] sum = new GeneralFunction[(int) floor(n / 2.) + 1];
+		for (int m = 0; m <= (int) Math.floor(n / 2.); m++) {
+			sum[m] = new Product(new Constant(constant(n, m)), new Pow(new Constant(n - 2 * m), new Variable(Settings.singleVariableDefault))).simplify();
+		}
+		GeneralFunction polynomial = new Sum(sum).simplify();
+		cache.put(n, polynomial);
+		return polynomial;
+	}
+
+	/**
+	 * Returns the nth Hermite polynomial with a variable of the specified String
+	 * @param n the nth polynomial
+	 * @param variable the String of the {@link Variable} of the polynomial
+	 * @return nth Hermite polynomial
+	 */
+	public static GeneralFunction hermitePolynomial(int n, String variable) {
+		GeneralFunction polynomial = hermitePolynomial(n);
+		if (Settings.singleVariableDefault.equals(variable))
+			return polynomial;
+		else {
+			Map<String, Variable> substitution = new HashMap<>() {
```

Comment:
if you're defining the HashMap, don't use an initializer like this; just say `new HashMap<>();` and run `put` on the next line